tests/testthat/test-infer_polars_dtype.R

```diff
@@ -0,0 +1,56 @@
+patrick::with_parameters_test_that(
+  "infer_polars_dtype() works for various objects",
+  x = {
+    skip_if_not_installed("hms")
+    skip_if_not_installed("blob")
+    skip_if_not_installed("bit64")
+    skip_if_not_installed("vctrs")
+    skip_if_not_installed("clock")
+
+    withr::with_timezone(
+      "UTC",
+      list2(
+        as_polars_series(1),
+        pl$DataFrame(a = 1L, b = "foo"),
```

Comment:
Include an example for LazyFrame too?

Reply:
Yes, ideally. But at the moment I can't add it here because `as_polars_series(<LazyFrame>)` fails.
Add a TODO comment.